pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -454,7 +457,14 @@ def register_plugins(cls, plugin_registry: Dict) -> None:
             find_unused_parameters=False,
         )
 
-    def _share_information_to_prevent_deadlock(self):
+    def _share_information_to_prevent_deadlock(self) -> None:
+        self._has_called_call_children_scripts = self.broadcast(self._has_called_call_children_scripts)
```

Comment:
let's do the broadcast directly after the call to `call_children_scripts()` so the value is actually meaningful from this point on.

Reply:
ok, can't do it directly after the call, but let's do it as early as possible once distributed is init.